scripts/tests/linux/log_line_processing.py

```diff
@@ -92,9 +94,14 @@ def __enter__(self):
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             text=True,
+            bufsize=1,  # Enable line buffering for immediate output from subprocess
         )
-        self.io_thread = threading.Thread(target=self._io_thread)
-        self.io_thread.start()
+        self.stdout_thread = threading.Thread(target=self._stdout_thread)
+        self.stderr_thread = threading.Thread(target=self._stderr_thread)
+        self.stdout_thread.start()
+        self.stderr_thread.start()
+        self.output_file = open(self.output_path, "wt", buffering=1)  # Enable line buffering
```

Comment:
open should be moved before the thread start, otherwise threads may _write to the file that is not yet open.

Reply:
Should I move it to before line 99?

Reply:
Moved it in the latest commit.